### PR TITLE
2.5.6 release notes

### PR DESCRIPTION
**Pull request by nathan-px**

### CODE REVIEW

File: content/reference/release-notes/portworx.md [hunk 1]
```diff
@@ -6,6 +6,30 @@ keywords: portworx, release notes
...
+| PWX-13173 | Added support for providing the `--vault-namespace` argument in pxctl commands: <br/><br/>setting cluster-wide secrets: <br/><br/>`/opt/pwx/bin/pxctl set-cluster-key --secret_options=vault-namespace=portworx px-vol` <br/><br/>create volume: `/opt/pwx/bin/pxctl volume create --secure --secret_key=pwx/custom-key --secret_options=vault-namespace=portworx px-vol` <br/><br/>host attach: `/opt/pwx/bin/pxctl host attach --secret_key=pwx/custom-key --secret_options=vault-namespace=portworx 57416092699073855` |
+
+### Fixes
+
+The following issues have been fixed:
+
+|**Issue Number**|**Issue Description**|
+|----|----|
+| PWX-14777 | If credentials were deleted from Kubernetes secrets while there were multiple pending references to the credentials in a Portworx cluster, Portworx generated an alert for every pending reference. <br/><br/>**User impact:** Users received an excessive number of alerts. <br/><br/> Portworx now generates only one alert every hour. Portworx, Inc. advises against deleting credentials if there are schedules managed by Stork. |
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:28]: @talakad

File: content/reference/release-notes/portworx.md [hunk 2]
```diff
@@ -6,6 +6,30 @@ keywords: portworx, release notes
...
+August 13, 2020
+
+### Improvements
+
+Portworx has upgraded or enhanced functionality in the following areas:
+
+| **Improvement Number** | **Improvement Description** |
+|----|----|
+| PWX-14845 | Added a new px install argument (VAULT_NAMESPACE) to set a global vault namespace for PX. This argument should be provided as a part of the kubernetes secret px-vault. | 
+| PWX-13173 | Added support for providing the `--vault-namespace` argument in pxctl commands: <br/><br/>setting cluster-wide secrets: <br/><br/>`/opt/pwx/bin/pxctl set-cluster-key --secret_options=vault-namespace=portworx px-vol` <br/><br/>create volume: `/opt/pwx/bin/pxctl volume create --secure --secret_key=pwx/custom-key --secret_options=vault-namespace=portworx px-vol` <br/><br/>host attach: `/opt/pwx/bin/pxctl host attach --secret_key=pwx/custom-key --secret_options=vault-namespace=portworx 57416092699073855` |
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:20]: @shailvipx

File: content/reference/release-notes/portworx.md [hunk 3]
```diff
@@ -6,6 +6,30 @@ keywords: portworx, release notes
...
+| PWX-14937 | Added a fix to allow pool expand using the add-disk operation on storage pools that have a journal partition. <br/><br/>**User impact:** If users have installed Portworx with cloud drive provisioning with auto journal (-j auto), pool expand with operation type add-disk will fail with error message saying it is not supported. The same failure will also be seen when users are using Autopilot to expand the pools. <br/><br/>**Resolution:** When pools are expanded using operation type add-disk, there is no actual limitation if Portworx was installed wit auto journal. The resolution was to remove the auto journal check if expand is being done using add-disk operation. |
+| PWX-15094 | A nil panic in Portworx sometimes occurred when Portworx successfully created a Vault client while Vault was in an error state.<br/><br/>**User impact:** Portworx returned an error: "too many open file handles" for subsequent vault operations.<br/><br/>**Resolution:** Portworx no longer panics when Vault is experiencing an error. |
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:31]: @adityadani

File: content/reference/release-notes/portworx.md (hunk 1 above)
Comment (talakad) [reply content/reference/release-notes/portworx.md:28]: Looks good to me.

File: content/reference/release-notes/portworx.md (hunk 3 above)
Comment (vilasd) [reply content/reference/release-notes/portworx.md:31]: We don't need to RL note PWX-15094

File: content/reference/release-notes/portworx.md (hunk 2 above)
Comment (adityadani) [inline content/reference/release-notes/portworx.md:20]:
```suggestion
| PWX-13173 | Added support for providing the `vault-namespace` argument in pxctl commands: <br/><br/>setting cluster-wide secrets: <br/><br/>`/opt/pwx/bin/pxctl set-cluster-key --secret_options=vault-namespace=portworx px-vol` <br/><br/>create volume: `/opt/pwx/bin/pxctl volume create --secure --secret_key=pwx/custom-key --secret_options=vault-namespace=portworx px-vol` <br/><br/>host attach: `/opt/pwx/bin/pxctl host attach --secret_key=pwx/custom-key --secret_options=vault-namespace=portworx 57416092699073855` |
```

File: content/reference/release-notes/portworx.md (hunk 3 above)
Comment (adityadani) [reply content/reference/release-notes/portworx.md:31]: I think we should release note this. This can happen on earlier PX versions.

Comment (vilasd) [reply content/reference/release-notes/portworx.md:31]: nope bud!! I don't want to just show our internal issues unless reported

File: content/reference/release-notes/portworx.md [hunk 4]
```diff
@@ -6,6 +6,30 @@ keywords: portworx, release notes
...
+| PWX-14937 | Added a fix to allow pool expand using the add-disk operation on storage pools that have a journal partition. <br/><br/>**User impact:** If users installed Portworx with cloud drive provisioning using the auto-journal (`-j auto`), expanding the pool with the `add-disk` operation type failed with a "not supported" error message. Users also saw the same error when using Autopilot to expand the pools. <br/><br/>**Resolution:** If Portworx was installed using the auto-journal, it can now expand pools using the `add-disk` operation type. |
+| PWX-15094 | A nil panic in Portworx sometimes occurred when Portworx successfully created a Vault client while Vault was in an error state.<br/><br/>**User impact:** Portworx returned an error: "too many open file handles" for subsequent vault operations.<br/><br/>**Resolution:** Portworx no longer panics when Vault is experiencing an error. |
```
Comment (vilasd) [inline content/reference/release-notes/portworx.md:31]: @adityadani is this good to remove ??? you still want to keep it ??

Comment (vilasd) [reply content/reference/release-notes/portworx.md:31]: You know what let it be :)

File: content/reference/release-notes/portworx.md [hunk 5]
```diff
@@ -6,6 +6,30 @@ keywords: portworx, release notes
...
+| PWX-14034 | Portworx did not read all the vault credentials for Kubernetes authentication methods provided in a Kubernetes secret. <br/><br/>**User impact:** Portworx Vault authentication failed.<br/><br/>**Resolution:** With 2.5.6, Portworx now reads all the Kubernetes authentication related input arguments from the Kubernetes secret and Portworx Vault authentication will succeed. | 
+| PWX-14937 | Added a fix to allow pool expand using the add-disk operation on storage pools that have a journal partition. <br/><br/>**User impact:** If users installed Portworx with cloud drive provisioning using the auto-journal (`-j auto`), expanding the pool with the `add-disk` operation type failed with a "not supported" error message. Users also saw the same error when using Autopilot to expand the pools. <br/><br/>**Resolution:** If Portworx was installed using the auto-journal, it can now expand pools using the `add-disk` operation type. |
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:30]: @harsh-px if you want to take one last look

File: content/reference/release-notes/portworx.md (hunk 2 above)
Comment (shailvipx) [reply content/reference/release-notes/portworx.md]: Needs to be changed to below. set-cluster-key does not take volume name at the end. 

| PWX-13173 | Added support for providing the `vault-namespace` argument in pxctl commands: <br/><br/>setting cluster-wide secrets: <br/><br/>`/opt/pwx/bin/pxctl set-cluster-key --secret_options=vault-namespace=portworx` <br/><br/>create volume: `/opt/pwx/bin/pxctl volume create --secure --secret_key=pwx/custom-key --secret_options=vault-namespace=portworx px-vol` <br/><br/>host attach: `/opt/pwx/bin/pxctl host attach --secret_key=pwx/custom-key --secret_options=vault-namespace=portworx 57416092699073855` |

File: content/reference/release-notes/portworx.md [hunk 6]
```diff
@@ -6,6 +6,30 @@ keywords: portworx, release notes
 series: release-notes
 ---
 
+## 2.5.6
+
+August 13, 2020
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:11]:
```suggestion
August 17, 2020
```

**Pull request merged**